Core/Src/main.c

```diff
@@ -95,19 +96,61 @@ int main(void)
   MX_ADC_Init();
   MX_TIM16_Init();
   MX_USART1_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
 
   /* USER CODE BEGIN 2 */
-  debug("\n\nBootcamp starting up...");
-  debug("Compiled on %s at %s", __DATE__, __TIME__);
 
+  //start pwm
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  //define buffers and variables
+  uint8_t spi_recieve_buf[10];
```

Comment:
These buffers are 10 bytes in length. 
The ADC sends only 10 bits of data (just a bit over 1 byte) in 3 bytes of message, so the data can be stored in a buffer of length 3, then the 10 bits of data extracted from that.
For the transmit message you have the 3 bytes you set later being transmitted, so a buffer length of 3 is enough.